ambry-cloud/src/main/java/com.github.ambry.cloud/VcrReplicationManager.java

```diff
@@ -242,6 +225,41 @@ void addPartition(PartitionId partitionId) throws ReplicationException {
     }
   }
 
+  /**
+   * Remove a replica of given {@link PartitionId} and its {@link RemoteReplicaInfo}s from the backup list.
+   * @param partitionId the {@link PartitionId} of the replica to removed.
+   * @throws ReplicationException if replicas initialization failed.
+   */
+  void removeReplica(PartitionId partitionId) throws ReplicationException {
+    PartitionInfo partitionInfo = partitionToPartitionInfo.remove(partitionId);
+    if (partitionInfo == null) {
+      logger.error("Partition {} not exist when remove from {}. ", partitionId, dataNodeId);
+      throw new ReplicationException("Partition not found");
+    }
+    removeRemoteReplicaInfoFromReplicaThread(partitionInfo.getRemoteReplicaInfos());
+    if (replicationConfig.replicationPersistTokenOnShutdownOrReplicaRemove) {
+      try {
+        persistor.write(partitionInfo.getLocalReplicaId().getMountPath(), false);
+      } catch (IOException | ReplicationException e) {
+        logger.error("Exception on token write when remove Partition {} from {}: ", partitionId, dataNodeId, e);
+        throw new ReplicationException("Exception on token write.");
+      }
+    }
+    Store cloudStore = partitionStoreMap.get(partitionId.toPathString());
+    if (cloudStore != null) {
+      try {
+        cloudStore.shutdown();
```

Comment:
Should the store be removed from `StoreManager` here or in some other place?
Also, should it be shut down using `StoreManager.shutdownBlobStore`?